src/base_alloc/base_alloc.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+
+#include "base_alloc.h"
+#include "base_alloc_internal.h"
+#include "utils_common.h"
+#include "utils_concurrency.h"
+
+// minimum size of a single pool of the base allocator,
+#define MINIMUM_POOL_SIZE (ba_os_get_page_size())
```

Comment:
should we have this documented somehow...? (perhaps we can do it, when BA is actually used)

Reply:
@lukaszstolarczuk base allocator wouldn't be public, similar as critnib

Reply:
@bratpiorka, yeah, I get that, but it will affect pools using this base

@ldorau, in the docs, I guess 😉

Reply:
I think it would be worth documenting if this would be exposed in cmake options or something. Now it isn't and we don't plan to do that. So documenting this inside a code as a comment is IMO enough.